linker/shared/src/main/scala/org/scalajs/linker/backend/javascript/Printers.scala

```diff
@@ -126,10 +126,31 @@ object Printers {
 
       case _ =>
         printIndent()
-        printTree(tree, isStat = true)
+        if (textuallyStartsWithObjectLiteral(tree)) {
+          print('(')
+          printTree(tree, isStat = false)
```

Comment:
`({ a: 1 }.a = 2)` is perfectly valid in JS. An assignment is an expression.

Reply:
Right... To me, the leap from "this statement starts with an object literal" to "this statement is a valid expression" is a bit big. From `textuallyStartsWithObjectLiteral` it seems that this is indeed the case, at least for our JS trees, but I do not understand whether (and if yes), why this would be the case.